project_cost_report_supply/report.py

```diff
@@ -53,50 +56,28 @@ def _get_shop_supply_categories(self, project, report_context):
             result.append(empty_category)
         return result
 
-    def _get_shop_supply_total(self, project):
-        """Get the total amount for the OUTSOURCING section.
-
-        :param project: the project.project record
-        :rtype: float
-        """
-        domain = self._get_shop_supply_analytic_line_domain(project)
-        result = self.env['account.analytic.line'].read_group(
-            domain=domain, fields=['account_id', 'amount'], groupby='account_id')
-        amount = result[0]['amount'] if result else 0
-        return float_round(-amount, 2)
-
-    def _get_rendering_variables(self, project, report_context):
-        """Add the variables related to the OUTSOURCING section."""
-        res = super()._get_rendering_variables(project, report_context)
-        shop_supply_total = self._get_shop_supply_total(project)
-        res.update({
-            'shop_supply_categories': self._get_shop_supply_categories(project, report_context),
-            'shop_supply_total': shop_supply_total,
-            'total_cost': float_round(res['total_cost'] + shop_supply_total, 2)
-        })
-        return res
-
-    @api.model
-    def get_foldable_categories(self, project_id):
-        """Add shop supply categories to foldable categories.
-
-        The SHOP SUPPLY section has only one category (the empty category).
-        """
-        res = super().get_foldable_categories(project_id)
-        res['shop_supply'] = [False]
-        return res
-
-    def _get_product_analytic_line_domain(self, project):
-        """Isolate the shop supply from product domain."""
-        domain = super()._get_product_analytic_line_domain(project)
-        return AND((domain, [('is_shop_supply', '=', False)]))
-
-    def _get_time_analytic_line_domain(self, project):
-        """Isolate the shop supply from time domain."""
-        domain = super()._get_time_analytic_line_domain(project)
-        return AND((domain, [('is_shop_supply', '=', False)]))
+    def _get_empty_shop_supply_category_label(self):
+        """Get the label to display on the unique shop supply category."""
+        return _('Shop Supply')
 
-    def _get_outsourcing_analytic_line_domain(self, project):
-        """Isolate the shop supply from outsourcing domain."""
-        domain = super()._get_outsourcing_analytic_line_domain(project)
-        return AND((domain, [('is_shop_supply', '=', False)]))
+    def _get_shop_supply_total(self, project):
+        lines = self._get_shop_supply_analytic_lines(project)
+        total_amount = sum(l.amount for l in lines)
+        return float_round(-total_amount, 2)
+
+    def _get_shop_supply_analytic_lines(self, project):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/numigi/odoo-project-addons/pullRequest?prid=4986859)